src/synology_dsm/api/photos/__init__.py

```diff
@@ -33,12 +33,12 @@ async def get_albums(
 
```

Comment:
I believe `"category": "normal_share_with_me"` needs to be added to the API call parameters so that shared albums are returned too. The default `normal` only returns albums already owned by the current user.

```python
        raw_data = await self._dsm.get(
            self.BROWSE_ALBUMS_API_KEY, "list", {
                "offset": offset, 
                "limit": limit, 
                "category": "normal_share_with_me"
            }
        )
```

Reply:
Thanks, that right! I forgot about that. But it seems that downloading items from a shared album owned by a different user is not allowed. I'm getting a 804 unknown error.
Thumbnail api is working though. Looking at the web client, download api is never used for the slideshow, and the download button is only visible on owned albums. So I think we can't do much about that.

Reply:
@alexismarquis In a use case like Synology shared album used as a source for screensavers(eg: https://github.com/j-a-n/lovelace-wallpanel#media-source-url), would this mean it will still not work? I would expect the download action to be executed by HA when screensaver is updated to show a new image.

Reply:
HA integration will be updated as well, and I think it will be better to use the thumbnail method with xl size, like the web app. Download method is there to get the original version of the picture.

Reply:
I was actually looking into this exact same thing last weekend. Unfortunately I'm out until Thursday. If you still haven't figured it out by then I can help investing.

You can use the dev tools in chrome to see what api calls Synology photos is doing. I didn't see any calls to download, it uses a different api to load the pictures themselves. My guess is that download would get you the original file, while the other call used by the front end may give you a transcoded browser-friendly version.